### PR TITLE
add auth test UT

### PR DESCRIPTION
**Pull request by Genius-pig**
add auth test UT

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.SET_STORAGE_GROUP, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.CREATE_TIMESERIES, user.getName()));
+
+    Assert.assertFalse(AuthorityChecker.check(user.getName(), Collections.singletonList(new PartialPath(nodeName)),
+        OperatorType.DELETE_TIMESERIES, user.getName()));
+  }
+}
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java:133]: It's not a good test. The user created has no privilege at all, so all assertions return false. The correct test method may be: 1. test the authority before granting, return false. 2. grant the privilege 3. test the authority again, return true.

File: server/src/test/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizerTest.java [hunk 2]
```diff
@@ -16,29 +16,37 @@
...
+import org.apache.iotdb.db.auth.entity.Role;
 import org.apache.iotdb.db.auth.entity.User;
 import org.apache.iotdb.db.conf.IoTDBConstant;
 import org.apache.iotdb.db.utils.EnvironmentUtils;
 import org.junit.After;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
 public class LocalFileAuthorizerTest {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizerTest.java:38]:
```suggestion
public class BasicAuthorizerTest {
```

Comment (Alima777) [reply server/src/test/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizerTest.java:38]: I think `BasicAuthorizerTest` is better..

File: server/src/test/java/org/apache/iotdb/db/auth/entity/UserTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+public class UserTest {
+
+  @Test
+  public void testUser() {
+    User user = new User("user", "password");
+    PathPrivilege pathPrivilege = new PathPrivilege("root.ln");
+    user.setPrivilegeList(Collections.singletonList(pathPrivilege));
+    Set<Integer> set = new HashSet<>();
+    set.add(1);
+    user.setPrivileges("root.ln", set);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/entity/UserTest.java:36]:
```suggestion
    user.setPrivileges("root.ln", Collections.singleton(1));
```

File: server/src/test/java/org/apache/iotdb/db/auth/entity/RoleTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+public class RoleTest {
+
+  @Test
+  public void testRole() {
+    Role role = new Role("role");
+    PathPrivilege pathPrivilege = new PathPrivilege("root.ln");
+    role.setPrivilegeList(Collections.singletonList(pathPrivilege));
+    Set<Integer> set = new HashSet<>();
+    set.add(1);
+    role.setPrivileges("root.ln", set);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/entity/RoleTest.java:36]:
```suggestion
    role.setPrivileges("root.ln", Collections.singleton(1));
```

File: server/src/test/java/org/apache/iotdb/db/auth/entity/PathPrivilegeTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+public class PathPrivilegeTest {
+
+  @Test
+  public void testPathPrivilege() {
+    PathPrivilege pathPrivilege = new PathPrivilege();
+    pathPrivilege.setPath("root.ln");
+    Set<Integer> set = new HashSet<>();
+    set.add(1);
+    pathPrivilege.setPrivileges(set);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/entity/PathPrivilegeTest.java:34]:
```suggestion
    pathPrivilege.setPrivileges("root.ln", Collections.singleton(1));
```

File: server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java [hunk 6]
```diff
@@ -32,49 +35,63 @@
...
+
+    OpenIdAuthorizer authorizer = new OpenIdAuthorizer(JSONObjectUtils.parse(OPEN_ID_PUBLIC_JWK));
+    boolean login = authorizer.login(jwt, null);
+
+    assertTrue(login);
+  }
+
+  @Test
+  public void isAdmin_hasAccess()
+      throws AuthException, ParseException {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java:53]:
```suggestion
  public void isAdmin_hasAccess() throws AuthException, ParseException {
```

File: server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java [hunk 7]
```diff
@@ -32,49 +35,63 @@
...
+
+    OpenIdAuthorizer authorizer = new OpenIdAuthorizer(JSONObjectUtils.parse(OPEN_ID_PUBLIC_JWK));
+    boolean admin = authorizer.isAdmin(jwt);
+
+    assertTrue(admin);
+  }
+
+  @Test
+  public void isAdmin_noAdminClaim()
+      throws AuthException, ParseException {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/auth/authorizer/OpenIdAuthorizerTest.java:65]:
```suggestion
  public void isAdmin_noAdminClaim() throws AuthException, ParseException {
```

**Review by Alima777 — CHANGES_REQUESTED** [review]
Try to modify all test named `test()` or add javadoc for it. The naming method is sooo ambiguous and not intuitive to maintain.

File: server/src/test/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizerTest.java (hunk 2 above)
Comment (Genius-pig) [reply server/src/test/java/org/apache/iotdb/db/auth/authorizer/LocalFileAuthorizerTest.java:38]: In fact, `BasicAuthorizer getInstance`  return a `LocalFileAuthorizer` instance

**Review by jt2594838 — APPROVED** [review]

File: server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  @Test
+  public void test() throws AuthException, IllegalPathException {
+    authorizer.createUser(user.getName(), user.getPassword());
+    authorizer.grantPrivilegeToUser(user.getName(), nodeName, 1);
+    PathPrivilege pathPrivilege = new PathPrivilege();
+    Set<Integer> set = new HashSet<>();
+    set.add(1);
+    pathPrivilege.setPrivileges(set);
```
[the remark this answers is not in the document]
Comment (Genius-pig) [reply server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java]: update

File: server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java (hunk 1 above)
Comment (Genius-pig) [reply server/src/test/java/org/apache/iotdb/db/auth/AuthorityCheckerTest.java]: update

**Comment by Alima777** [comment]
Hi, check the CI please.

**Review by Alima777 — APPROVED** [review]

**Pull request merged**